src/gui/window.rs

```diff
@@ -131,21 +304,124 @@ impl GuiWasabiWindow {
-                    self.render_scene
-                        .draw(state, ui, &key_view, &mut self.midi_file, 0.25);
-                render_result_data = Some(result);
-            });
+                if let Some(midi_file) = self.midi_file.as_mut() {
+                    let one_sec = Duration::from_secs(1);
+                    let time = midi_file.timer().get_time();
+                    let events = ui.input().events.clone();
+                    for event in &events {
+                        if let egui::Event::Key { key, pressed, .. } = event {
+                            if pressed == &true {
+                                match key {
+                                    egui::Key::ArrowRight => {
+                                        midi_file.timer_mut().seek(time + one_sec)
+                                    }
+                                    egui::Key::ArrowLeft => {
+                                        midi_file.timer_mut().seek(time - one_sec)
+                                    }
+                                    egui::Key::Space => midi_file.timer_mut().toggle_pause(),
+                                    _ => {}
+                                }
+                            }
+                        }
+                    }
 
-        let render_result_data = render_result_data.unwrap();
+                    let result = self
+                        .render_scene
+                        .draw(state, ui, &key_view, midi_file, note_speed);
+                    stats.notes_on_screen = result.notes_rendered;
+                    render_result_data = Some(result);
+                }
+            });
 
         // Render the keyboard
         egui::TopBottomPanel::top("Keyboard panel")
             .height_range(keyboard_height..=keyboard_height)
             .frame(no_frame)
             .show(&ctx, |ui| {
+                let events = ui.input().events.clone();
+                for event in &events {
+                    if let egui::Event::Key { key, pressed, .. } = event {
+                        if pressed == &true {
+                            match key {
+                                egui::Key::F => match temp_settings.panel_visible {
+                                    true => temp_settings.panel_visible = false,
+                                    false => temp_settings.panel_visible = true,
+                                },
+                                egui::Key::G => match temp_settings.stats_visible {
+                                    true => temp_settings.stats_visible = false,
+                                    false => temp_settings.stats_visible = true,
+                                },
+                                _ => {}
+                            }
+                        }
+                    }
+                }
+
+                let colors = if self.midi_file.is_some() {
+                    render_result_data.unwrap().key_colors
```

Comment:
Ideally it's good to avoid `.unwrap()` as much as possible. Yes technically this wouldn't crash, but it's not guaranteed, so instead it's more standard to do:
```rs
let colors = if let Some(data) = render_result_data {
    data.key_colors
} else ....
```